packages/std/src/imports.rs

```diff
@@ -54,39 +54,23 @@ impl ExternalStorage {
     pub fn new() -> ExternalStorage {
         ExternalStorage {}
     }
+}
 
-    pub fn get_with_result_length(
-        &self,
-        key: &[u8],
-        result_length: usize,
-    ) -> StdResult<Option<Vec<u8>>> {
+impl ReadonlyStorage for ExternalStorage {
+    fn get(&self, key: &[u8]) -> StdResult<Option<Vec<u8>>> {
         let key = build_region(key);
         let key_ptr = &*key as *const Region as *const c_void;
-        let value_ptr = alloc(result_length);
 
-        let read = unsafe { db_read(key_ptr, value_ptr) };
-        if read == -1_000_001 {
-            return Err(generic_err("Allocated memory too small to hold the database value for the given key. \
-                You can specify custom result buffer lengths by using ExternalStorage.get_with_result_length explicitely."));
-        } else if read == -1_001_001 {
+        let read = unsafe { db_read(key_ptr) };
+        if read == 0 {
             // key does not exist in external storage
             return Ok(None);
-        } else if read < 0 {
-            return Err(generic_err(format!(
-                "Error reading from database. Error code: {}",
-                read
-            )));
         }
 
+        let value_ptr = read as *mut c_void;
         let data = unsafe { consume_region(value_ptr) }?;
```

Comment:
Okay, I now check both pointers in `consume_region` for null, the input `ptr` and the pointer the Region points to. But those cases are now considered horrible bugs that trigger a panic instead of returning a Result we could recover from. That makes all usages of `consume_region` simpler.

As a result, ResdonlyStorage::get does not need the result anymore 🎉

Reply:
awesome! :muscle: